File: languages/python/problem-xn.py
#!/usr/bin/env python3
#
# Find the minimum natural number that increases N times after
# moving its last digit N to the beginning of the number.
#
for n in range(2, 10):
    for k in range(1, 100):
        x = n * (10**k - n)
        if x % (10*n - 1) == 0:
            v = 10 * x // (10*n - 1) + n
            print("For", n, "solution =", v)
            break

File: languages/haskell/problem-xn.hs
#!/usr/bin/env runhaskell

module Main where

solve n 1 = 1
solve n x = d + 10 * (solve n (x `div` 10 + d * n))
            where d = x `mod` 10

main = do
	print (solve 2 2)
	print (solve 3 3)
	print (solve 4 4)
	print (solve 5 5)
	print (solve 6 6)
	print (solve 7 7)
	print (solve 8 8)
	print (solve 9 9)
